src/ApiCallable.php

```diff
@@ -206,18 +180,24 @@ private static function setCustomHeader($callable, $headerDescriptor, $userHeade
      * @throws ValidationException
      * @return callable
      */
-    public static function createApiCall($stub, $methodName, CallSettings $settings, $options = [])
-    {
+    public static function createApiCall(
+        TransportInterface $transport,
+        $methodName,
+        CallSettings $settings,
+        $options = []
+    ) {
         ApiCallable::validateApiCallSettings($settings, $options);
 
-        $callable = array($stub, $methodName);
-        if (array_key_exists('grpcStreamingDescriptor', $options)) {
-            $apiCall = ApiCallable::createGrpcStreamingApiCall(
-                $callable,
-                $options['grpcStreamingDescriptor']
-            );
-        } else {
-            $apiCall = ApiCallable::createUnaryApiCall($callable);
+        $apiCall = function() use ($transport, $methodName) {
+            $callable = [$transport, $methodName];
+            return call_user_func_array($callable, func_get_args());
+        };
+
+        // Call the sync method "wait" if this is not a gRPC call
+        if (!array_key_exists('grpcStreamingDescriptor', $options)) {
```

Comment:
I opted to use the `wait` function only when the gRPC streaming descriptors aren't set. We could make this generic, but they _are_ gRPC specific, so I'm not overly concerned with it. I am definitely open to better ways to go about this!

Reply:
**Note, this has been refactored out in #109 and looks way better**